code/modules/atmospherics/machinery/components/unary_devices/cryo.dm

```diff
@@ -268,14 +266,9 @@ GLOBAL_VAR_INIT(cryo_overlay_cover_off, mutable_appearance('icons/obj/cryogenics
 			mob_occupant.Sleeping((mob_occupant.bodytemperature * sleep_factor) * 1000 * delta_time)
 			mob_occupant.Unconscious((mob_occupant.bodytemperature * unconscious_factor) * 1000 * delta_time)
 		if(beaker)
-			if(reagent_transfer == 0) // Magically transfer reagents. Because cryo magic.
-				beaker.reagents.trans_to(occupant, CRYO_TX_QTY * delta_time, efficiency * CRYO_MULTIPLY_FACTOR, methods = VAPOR) // Transfer reagents.
-				consume_gas = TRUE
-			reagent_transfer += 1
-			if(reagent_transfer >= CRYO_THROTTLE_CTR_MAX * efficiency) // Throttle reagent transfer (higher efficiency will transfer the same amount but consume less from the beaker).
-				reagent_transfer = 0
-
-	return 1
+			beaker.reagents.trans_to(occupant, (CRYO_TX_QTY / (efficiency * CRYO_MULTIPLY_FACTOR)) * delta_time, efficiency * CRYO_MULTIPLY_FACTOR, methods = VAPOR) // Transfer reagents.
```

Comment:
This means it injects the whole amount of the reagent in the beaker, but only uses a small portion of it. Right? Since the second value is a multiplier, the math looks like
`(QTY / (efficiency )) * efficiency`

Am I right in this/does this behave how you excepted? 
Oh and how does this formula change when deltatime raises/loweres

Reply:
My goal was that there is a set constant number injected per delta time, so it would inject say 0.5 units for delta time that passed. 
You are perfectly right about the math, i moved the efficiency to the multiplier input on the proc, so it draws less reagent per upgrade while adding the multiplier.

I suppose a test case might illustrate this better:
Lets say that delta_time is buttery smooth at a constant 1 for the whole operation.
A patient gets in, beaker is drawn according to `CRYO_TX_QTY / (efficiency * CRYO_MULTIPLY_FACTOR)`, for a level 1 matter bin this is around 0.02. Then the 0.02 unit of reagents is multiplied according to `efficiency * CRYO_MULTIPLY_FACTOR` which would mean that 0.5 units is injected into the patient. Upgrades to the cryo would preserve the reagent transfer rate (0.5) per delta time while lowering the draw rate (0.02, 0.01, 0.007, and 0.005 for each matter bin tier.)

Reply:
Also, I'm sure you know this already, but for the sake of clarity the reagent multiplication rate (25,50,75,100) per upgrade is the exact same as the old one (both using `efficiency * CRYO_MULTIPLY_FACTOR`).

Reply:
As for the behaviour with `delta_time`, I must admit that I'm not very familiar with how the controller works. The old code have the reagent transfer rate multiplied with `delta_time` so i kept it there.

Reply:
Ah I'm braindead, I thought this was operating on the whole volume of the beaker. This is fine.

As far as delta time goes, it's a static number calculated based on the wait of the processing subsystem, which in theory should never change. As it stands currently it only exists to make changing those waits easier.

In this case, it's value is 2 since SSobj fires every 2 seconds. All that's important is the amount of reagent that's moved every second is consistent. So if delta time changes, the amount of reagent doesn't change. This should be fine